sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -503,13 +503,32 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
     }
   }
 
+  private def convertTypedLiteralToString(literal: Literal): String = literal match {
+    case Literal(data: Int, _: DateType) =>
+      UTF8String.fromString(
```

Comment:
Could you clarify why do you need the conversion: String -> UTF8String -> String

Reply:
> Could you clarify why do you need the conversion: String -> UTF8String -> String

Forgot where I see similar logical, emmm, so this place don't need this convert to keep standard charset UTF8